services/supervisor/tests/supervisor/migrations/migration_service.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is adapted version of service with migrations from `testkit/tests/migrations/mod.rs`.
+//
+// While version in `testkit` is designed for testing the migration process itself,
+// this version is designed to test migration workflow in supervisor service.
+//
+// As a result, this version is somewhat simplified and does not test big/random data sets,
+// but focuses more on different migration scenarios.
+
+// cspell:ignore Trillian, Vogon
+
+use exonum::{
+    merkledb::{
+        access::{Access, FromAccess},
+        ProofEntry,
+    },
+    runtime::{
+        migrations::{
+            InitMigrationError, LinearMigrations, MigrateData, MigrationContext, MigrationError,
+            MigrationScript,
+        },
+        versioning::Version,
+        ExecutionContext, ExecutionError, InstanceId,
+    },
+};
+use exonum_derive::*;
+use exonum_rust_runtime::{DefaultInstance, Service, ServiceFactory};
+
+#[derive(Debug, FromAccess)]
+pub struct Schema<T: Access> {
+    pub entry: ProofEntry<T::Base, u8>,
+}
+
+impl<T: Access> Schema<T> {
+    pub fn new(access: T) -> Self {
+        Self::from_root(access).unwrap()
+    }
+}
+
+/// Initial service schema.
+pub(super) mod v01 {
```

Comment:
Nit: IMO, `pub(super)`  qualifiers can be elided to `pub` given that it's test code. Also, since `ENTRY_VALUE` constants are seemingly used only in this module, you could use `pub(super)` for them.